CriticalMass/AnnotationController.swift

```diff
@@ -19,4 +19,25 @@ class AnnotationController<T: IdentifiableAnnnotation, K: MKAnnotationView> {
     }
 
     open func setup() {}
+
+    /// This method updates the displayed Annotations of the annotationType T. Annotations that already have a AnnotationView are updated with a new location, Annotations that don't have a View yet, will be added, and previously displayed annotations of the given type will be removed from the Map
+    /// - Parameter locations: A dictionary mapping between  identifier and their locations
+    open func updateAnnotations(locations: [String: Location]) {
+        var unmatchedLocations = locations
+        var unmatchedAnnotations: [MKAnnotation] = []
+        // update existing annotations
+        mapView.annotations.compactMap { $0 as? T }.forEach { annotation in
+            if let location = unmatchedLocations[annotation.identifier] {
+                annotation.location = location
+                unmatchedLocations.removeValue(forKey: annotation.identifier)
+            } else {
+                unmatchedAnnotations.append(annotation)
+            }
+        }
+        let annotations = unmatchedLocations.map { T(location: $0.value, identifier: $0.key) }
```

Comment:
Would every AnnotationController need to implement `IdentifiableAnnnotation` which is a subclass of `MKPointAnnotation`? Since `T: IdentifiableAnnnotation`

Reply:
Yes we need identifiable annotations to be able to perform diffing operations on displayed annotations

Reply:
This specific line might need to change because there is currently no way to inject data

Reply:
I added a mechanism to inject data into Annotations if needed